pkgs/top-level/perl-packages.nix

```diff
@@ -11353,16 +11353,13 @@ let
 
   LaTeXML = buildPerlPackage rec {
     pname = "LaTeXML";
-    version = "0.8.5";
+    version = "0.8.6";
     src = fetchurl {
       url = "mirror://cpan/authors/id/B/BR/BRMILLER/${pname}-${version}.tar.gz";
-      sha256 = "0dr69rgl4si9i9ww1r4dc7apgb7y6f7ih808w4g0924cvz823s0x";
+      sha256 = "1ccvdq7asxq6iw8x8ihwf5xs2mp7fkwm467xy7g8spkznr8wcacm";
     };
     outputs = [ "out" "tex" ];
-    propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageSize JSONXS LWP ParseRecDescent ImageMagick PodParser TextUnidecode XMLLibXSLT ];
-    preCheck = ''
-      rm t/931_epub.t # epub test fails
```

Comment:
Still fails

Reply:
Oh. Apparently the error is
```
Error in tempdir() using /build/XXXXXXXXXX: Parent directory (/build) is not safe (Directory owned neither by root nor the current user) at /build/LaTeXML-0.8.6/blib/script/../lib/LaTeXML.pm line 196.
```
I'll remove the test as I don't see an obvious way to handle this.

Reply:
This works
```diff
diff --git a/pkgs/top-level/perl-packages.nix b/pkgs/top-level/perl-packages.nix
index 6729ed28bc8..6b74bf81c0c 100644
--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -11360,8 +11360,11 @@ let
     };
     outputs = [ "out" "tex" ];
     propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageMagick ImageSize JSONXS LWP ParseRecDescent PodParser TextUnidecode XMLLibXSLT ];
-    preCheck = /* epub test fails: File::Temp considers /build insecure */ ''
-      rm t/931_epub.t
+    # https://github.com/brucemiller/LaTeXML/issues/1669
+    preCheck = ''
+      for file in ./lib/LaTeXML.pm ./lib/LaTeXML/Post/LaTeXImages.pm ./blib/lib/LaTeXML/Post/LaTeXImages.pm ./blib/lib/LaTeXML.pm; do
+        substituteInPlace "$file" --replace "File::Temp::HIGH" "File::Temp::MEDIUM"
+      done
     '';
     nativeBuildInputs = [ pkgs.makeWrapper ] ++ lib.optional stdenv.isDarwin shortenPerlShebang;
     makeMakerFlags = "TEXMF=\${tex} NOMKTEXLSR";
```

Reply:
Thanks for the investigation @veprbl! I suggest we keep the current `rm t/931_epub.t` (with link to the new upstream issue) and backport the `File::Temp::MEDIUM` patch once upstream makes a move. Deviating from upstream on a security setting does not feel right.

Reply:
I don't think there is any security going on, see the 22y old bug in File::Temp. But I don't mind keeping the rm.